### ISSUE TITLE
wx backends don't send close_event if window is closed via "q" keypress

### ISSUE THREAD

**Issue by anntzer** [matplotlib/matplotlib#16919]
### Bug report

**Bug summary**

Connect a callback to the "close_event" of a wx(agg) window, and close the window using the "q" shortcut.  The event is not emitted (unlike for qt5, gtk3, and tk).

**Code for reproduction**

```
$ MPLBACKEND=wxagg python -c 'from pylab import *; gcf().canvas.mpl_connect("close_event", print); show()'
```
and press 'q'.

**Actual outcome**

Nothing printed.

**Expected outcome**

A CloseEvent printed (as for the other GUI toolkits).

**Matplotlib version**

  * Operating system: linux
  * Matplotlib version: master
  * Matplotlib backend (`print(matplotlib.get_backend())`): wxagg
  * Python version: 38
  * Jupyter version (if applicable): no
  * Other libraries: 

attn @RobinD42 perhaps you may have an idea?

**Comment by RobinD42** [comment]
It looks like it is due to `FigureManagerWx.destroy` calling `self.frame.Destroy` instead of `self.frame.Close`. If I make that change and also change `FigureFrameWx._onClose` to check if it still exists before calling `Destroy`, like the following, then it works for me.

```
    def _onClose(self, evt):
        DEBUG_MSG("onClose()", 1, self)
        self.canvas.close_event()
        self.canvas.stop_event_loop()
        Gcf.destroy(self.num)
        if self:
            self.Destroy()
```

NOTE: I only tested this in a couple super simple examples using wxPython4, so I don't know if this might cause any regressions elsewhere.

**Comment by anntzer** [comment]
thanks for the quick reply; I included your fix in https://github.com/matplotlib/matplotlib/pull/16917.

**Comment by anntzer** [comment]
Closed by #16917
